ostd/src/sync/rcu/mod.rs

```diff
@@ -371,13 +371,13 @@ impl<P: OwnerPtr> RcuOptionReadGuard<'_, P> {
 /// The pointer must be previously returned by `into_raw` and the pointer
 /// must be only be dropped once.
 unsafe fn delay_drop<P: OwnerPtr>(pointer: NonNull<<P as OwnerPtr>::Target>) {
-    struct ForceSend<P: OwnerPtr>(NonNull<<P as OwnerPtr>::Target>);
+    struct ForceSend(*mut ());
```

Comment:
Assuming there is no soundness problems. Here, we should use `NonNull<()>` instead. See [`NonNull::cast`](https://doc.rust-lang.org/std/ptr/struct.NonNull.html#method.cast).